projects/canopy/src/lib/repsonsive/hide-at/hide-at.directive.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, DebugElement, Input } from '@angular/core';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+import { LgHideAtDirective } from './hide-at.directive';
+
+@Component({
+  template: ` <div [lgHideAt]="lgHideAt ? lgHideAt : undefined">How you see me...</div> `,
```

Comment:
I think this could be written as `[lgHideAt]="lgHideAt || null"`.

I would use `null` instead of `undefined` because that's the pattern used when handling attributes in angular.

The problem with directives is that they don't handle logic like attributes, so doing:
```
lgHideAt="null"
```
will not translate in the directive not being added like for a regular attribute.

Instead you should add the logic to say if the `responsiveUtilClass` is defined then add the new class otherwise do nothing.

If you check your code as it is now you most probably will see that if you set the `lgHideAt` as `undefined` or `null` the code will still run and add `lg-hide-at--undefined` (which works simply because that class doesn't exist).

Reply:
Thanks for spotting that @elenagarrone - good call! 

I've update the code and added a test for it too, let me know what you think.